pkgs/top-level/all-packages.nix

```diff
@@ -4554,6 +4554,7 @@ in
 
   lalezar-fonts = callPackage ../data/fonts/lalezar-fonts { };
 
+  ldcBootstrap = callPackage ../development/compilers/ldc/bootstrap.nix { };
```

Comment:
I am not sure if it is good to provide the bootstrap package this way.
It should not be available to the outside because it is not a "nice" package but a hackish one and is only needed for building the real package which provides the same functionality.
I can only imagine a testing scenario where having the bootstrap package available that way but this is highly speculative as I don't have a concrete example.
Even if such an example could be found it's for sure quite rare and can be added by a developer temporarily without polluting the namespace.

Reply:
I agree, `ldcBootstrap` is an implementation detail that shouldn't be exposed.

Reply:
Could advertise it as `ldc_bin` or `ldc.bootstrap`, probably.

In any case, it should be exposed in some way, otherwise how one can override it.

Reply:
@veprbl What do you mean?

It could simply be done by calling binary.nix directly from generic.nix like it was done in the previous solution [1].
This way bootstrap.nix and ldcBootstrap in all-packages.nix would not be necessary any more.

[1] https://github.com/NixOS/nixpkgs/blob/19.09/pkgs/development/compilers/ldc/default.nix#L13

Reply:
removed.

Reply:
@ThomasMader If I need to override bootstrap ldc in an overlay, how do I do that?

Reply:
@veprbl I see, @lionello already solved it by using callPackage as the default parameter like this -> ldcBootstrap ? callPackage ./bootstrap.nix { }

Reply:
Yes, this looks better. It would also help if the original bootstrap value was exposed. e.g. add `passthru = { inherit ldcBootstrap; }`.